packages/components/select/src/option-group.vue

```diff
@@ -57,18 +60,22 @@ export default defineComponent({
       children.value.some((option) => option.visible === true)
     )
 
-    const isOption = (node) =>
-      node.type?.name === 'ElOption' && !!node.component?.proxy
+    const isOption = (
+      node: VNode
+    ): node is VNode & { component: OptionInternalInstance } =>
+      (node.type as Component).name === 'ElOption' && !!node.component?.proxy
```

Comment:
Can the optional chaining be removed here? Can we be sure that type exists?

Reply:
I think it's ok because the type of `node.type` is `string | VNode | Component | typeof Text_2 | typeof Static | typeof Comment_2 | typeof Fragment | typeof TeleportImpl | typeof SuspenseImpl`, not including nullish.